### PR TITLE
Google Places Autocomplete

### PR DESCRIPTION
**Pull request by henryhu1**
for addresses when creating listing

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt [hunk 1]
```diff
@@ -16,27 +27,62 @@ import org.uwaterloo.subletr.api.apis.ListingsApi
...
+	private val bounds: RectangularBounds = RectangularBounds.newInstance(
+		LatLng(UWATERLOO_LATITUDE_DOUBLE, UWATERLOO_LONGITUDE_DOUBLE),
+		LatLng(UWATERLOO_LATITUDE_DOUBLE, UWATERLOO_LONGITUDE_DOUBLE)
+	)
+	private val placesClient: PlacesClient = Places.createClient(applicationContext)
 
 	val fullAddressStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
+	private val debouncedAddressStream: Observable<String> = fullAddressStream
+		.debounce(1, TimeUnit.SECONDS, Schedulers.computation())
+	private val addressAutocompleteStream: Observable<ArrayList<AutocompletePrediction>> = debouncedAddressStream
```
Comment (henryhu1) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: it works and I don't see performance issues but if something's not done right... I'll fix it

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt [hunk 2]
```diff
@@ -16,22 +27,56 @@ import org.uwaterloo.subletr.api.apis.ListingsApi
...
+					val predictions = ArrayList<AutocompletePrediction>()
+					for (prediction in response.autocompletePredictions) {
+						predictions.add(prediction)
+					}
+					emitter.onNext(predictions)
+				}.addOnFailureListener { exception: Exception? ->
+					emitter.onError(exception ?: RuntimeException())
+				}
+		}
+	}
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: I think this should be subscribed on an IO thread?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt (hunk 1 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: Seems fine. I think you can combine this stream with the `debouncedAddressStream`.
I don't see currently why they need to be separate streams.

**Review by leeminkyu16 — APPROVED** [review]
Let's Gamble, Try Merging

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt (hunk 1 above)
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: combined

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt (hunk 2 above)
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: subscribed

**Review by abhaykshar — COMMENTED** [review]
is this just a visual thing on my emulator or does this need to be the same size as the search bar:
image

**Comment by henryhu1** [comment]
> is this just a visual thing on my emulator or does this need to be the same size as the search bar: image

nah the dropdown isn't the same size. I tried to fixing it but the approach I had made the dropdown un-droppable so I left it as is.

**Review by abhaykshar — APPROVED** [review]

**Pull request merged**